app/services/moab_validator.rb

```diff
@@ -1,28 +1,45 @@
 # frozen_string_literal: true
 
 ##
-# mixin with methods for running StorageObjectValidator
-module MoabValidationHandler
-  # expects the class that includes this module to have the following methods:
+# service class with methods for running StorageObjectValidator
+class MoabValidator
+  attr_reader :druid, :storage_location, :results, :caller_validates_checksums
+
   # #druid - String (the "bare" druid, e.g. 'ab123cd4567', sans 'druid:' prefix)
   # #storage_location - String - the root directory holding the druid tree (the storage root path)
   # #results - AuditResults - the instance the including class is using to track findings of interest
-  # #complete_moab - CompleteMoab - instance of the complete moab being validated
+  # #complete_moab - CompleteMoab - optional - instance of the complete moab being validated, if already available from the caller
+  # #caller_validates_checksums - was this called by code that re-computes checksums to confirm that they match the values listed in the manifests?
+  def initialize(druid:, storage_location:, results:, complete_moab: nil, caller_validates_checksums: false)
+    @druid = druid
+    @storage_location = storage_location
+    @results = results
+    @complete_moab = complete_moab
+    @caller_validates_checksums = caller_validates_checksums
+  end
 
   def object_dir
     @object_dir ||= "#{storage_location}/#{DruidTools::Druid.new(druid).tree.join('/')}"
   end
 
-  def moab
-    @moab ||= Moab::StorageObject.new(druid, object_dir)
+  def complete_moab
```

Comment:
I wonder if you are open to renaming this method to disentangle it from the possibly injected `complete_moab`. For instance, could line 17 be `@complete_moab = complete_moab || default_complete_moab` with this method renamed to `#default_complete_moab`? (I don't love the name; it's a placeholder for demonstrating my point.) Or is it not possible to settle on the value of the complete_moab at initialization time?

Reply:
ah yeah... some callers already have the `CompleteMoab` instance available, and allowing them to provide it allowed less code churn and more incremental refactoring (as well as seeming more efficient in terms of reducing DB lookups).  but i do like the idea of disentangling this a bit more.

slight worry that changing when the moab is looked up will result in more test churn, but if that ends up exposing overly brittle tests, and isn't too much of a pain to fix, i'll go ahead and fix that too (though also maybe in a separate PR, to keep things easier to review, will see how it plays out).

Reply:
@jmartin-sul OK, sounds good to me.

Reply:
ok, so i tried this, and similar story to getting `druid` from `results`:  it breaks a bunch of existing tests to look up the `CompleteMoab` earlier.  many of these are likely more brittle than they need to be to accurately test what they're looking for.  but i'd prefer to ticket this suggestion as follow on work, if that's ok.